src/policy/largeobjectspace.rs

```diff
@@ -227,46 +226,12 @@ impl<VM: VMBinding> LargeObjectSpace<VM> {
         true
     }
 
-    fn test_mark_bit(&self, object: ObjectReference, value: usize) -> bool {
-        Self::read_gc_word(object) & MARK_BIT == value
+    fn test_mark_bit(&self, object: ObjectReference, value: u8) -> bool {
+        GCByte::read::<VM>(object) & MARK_BIT == value
     }
 
     fn is_in_nursery(&self, object: ObjectReference) -> bool {
-        Self::read_gc_word(object) & NURSERY_BIT == NURSERY_BIT
-    }
-
-    fn read_gc_word(o: ObjectReference) -> usize {
-        if USE_PRECEEDING_GC_HEADER {
-            unsafe {
-                (VM::VMObjectModel::object_start_ref(o) - PRECEEDING_GC_HEADER_BYTES)
-                    .load::<usize>()
-            }
-        } else {
-            VM::VMObjectModel::read_available_bits_word(o)
-        }
-    }
-
-    fn write_gc_word(o: ObjectReference, value: usize) {
-        if USE_PRECEEDING_GC_HEADER {
-            unsafe {
-                (VM::VMObjectModel::object_start_ref(o) - PRECEEDING_GC_HEADER_BYTES)
-                    .store::<usize>(value)
-            };
-        } else {
-            VM::VMObjectModel::write_available_bits_word(o, value);
-        }
-    }
-
-    fn attempt_gc_word(o: ObjectReference, old: usize, new: usize) -> bool {
-        if USE_PRECEEDING_GC_HEADER {
-            unsafe {
-                (VM::VMObjectModel::object_start_ref(o) - PRECEEDING_GC_HEADER_BYTES)
-                    .compare_exchange::<AtomicUsize>(old, new, Ordering::SeqCst, Ordering::SeqCst)
-                    .is_ok()
-            }
-        } else {
-            VM::VMObjectModel::attempt_available_bits(o, old, new)
-        }
+        GCByte::read::<VM>(object) & NURSERY_BIT == NURSERY_BIT
```

Comment:
I think for OpenJDK this is always false. HotSpot will crash if we try to keep extra metadata in object headers during mutator time. We can fix this in the future after we use side metadata for OpenJDK.